api/pkg/service/catalog/catalog.go

```diff
@@ -106,6 +107,50 @@ func (s *service) RefreshAll(ctx context.Context, p *catalog.RefreshAllPayload)
 	return res, nil
 }
 
+// List the all type of errors of a catalog
+func (s *service) CatalogError(ctx context.Context, p *catalog.CatalogErrorPayload) (*catalog.CatalogErrorResult, error) {
+
+	log := s.Logger(ctx)
+	db := s.DB(ctx)
+
+	ctg := model.Catalog{}
+	if err := db.Where(&model.Catalog{Name: p.CatalogName}).Model(&model.Catalog{}).First(&ctg).Error; err != nil {
```

Comment:
`Model(&model.Catalog{})` is not required I guess. can you pls check without it does the query executes the same?